tests/e2e/tests/build/build-app-shell.ts

```diff
@@ -0,0 +1,142 @@
+
+      import { AppComponent } from './app.component';
+
+      @NgModule({
+        imports: [
+          BrowserModule.withServerTransition({ appId: 'appshell-play' }),
+          RouterModule
+        ],
+        declarations: [AppComponent],
+        bootstrap: [AppComponent]
+      })
+      export class AppModule { }
+    `))
+    .then(() => writeFile('src/app/app.component.html', stripIndent`
+      Hello World
+      <router-outlet></router-outlet>
+    `))
+    .then(() => writeFile('src/tsconfig.server.json', stripIndent`
+      {
+        "extends": "../tsconfig.json",
+        "compilerOptions": {
+          "outDir": "../out-tsc/app",
+          "baseUrl": "./",
+          "module": "commonjs",
+          "types": []
+        },
+        "exclude": [
+          "test.ts",
+          "**/*.spec.ts"
+        ],
+        "angularCompilerOptions": {
+          "entryModule": "app/app.server.module#AppServerModule"
+        }
+      }
+    `))
+    .then(() => writeFile('src/main.server.ts', stripIndent`
+      export {AppServerModule} from './app/app.server.module';
+    `))
+    .then(() => writeFile('src/app/app.server.module.ts', stripIndent`
+      import {NgModule} from '@angular/core';
+      import {ServerModule} from '@angular/platform-server';
+      import { Routes, RouterModule } from '@angular/router';
+
+      import { AppModule } from './app.module';
+      import { AppComponent } from './app.component';
+      import { ShellComponent } from './shell.component';
+
+      const routes: Routes = [
+        { path: 'shell', component: ShellComponent }
+      ];
+
+      @NgModule({
+        imports: [
+          // The AppServerModule should import your AppModule followed
+          // by the ServerModule from @angular/platform-server.
+          AppModule,
+          ServerModule,
+          RouterModule.forRoot(routes),
+        ],
+        // Since the bootstrapped component is not inherited from your
+        // imported AppModule, it needs to be repeated here.
+        bootstrap: [AppComponent],
+        declarations: [ShellComponent],
+      })
+      export class AppServerModule {}
+    `))
+    .then(() => writeFile('src/app/shell.component.ts', stripIndent`
+      import { Component } from '@angular/core';
+      @Component({
+        selector: 'app-shell',
+        template: '<p>shell Works!</p>',
+        styles: []
+      })
+      export class ShellComponent {}
+    `))
+    .then(() => updateJsonFile('package.json', packageJson => {
+      const dependencies = packageJson['dependencies'];
+      dependencies['@angular/platform-server'] = platformServerVersion;
+    })
+    .then(() => npm('install')))
+    .then(() => ng('build', '--prod'))
+    .then(() => expectFileToMatch('dist/index.html', /shell Works!/))
+    .then(() => ng('build', '--prod', '--skip-app-shell'))
+    .then(() => expectToFail(() => expectFileToMatch('dist/index.html', /shell Works!/)));
```

Comment:
I have seen issues with cross-talk between tests for the Service Worker - perhaps we need to clean up `.angular-cli.json` here?

Reply:
I'm not sure what you're referring to here. I think making app shell and service worker work well together should be a priority.

Reply:
I think he means this : angular/angular#20360

Just tested AppShell from CLI 1.6.0-rc.0 and it's failing because of this.